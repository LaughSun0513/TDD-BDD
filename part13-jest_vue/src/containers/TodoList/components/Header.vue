<template>
	<div class="header-box">
		<div class="header-content">
			TodoList
			<input
				class="header-input"
				data-test="input"
				v-model="inputValue"
				@keyup.enter="addTodoItem"
				placeholder="请输入待办事项"
			/>
		</div>
	</div>
</template>

<script>
export default {
	name: 'Header',
	data() {
		return {
			inputValue: ''
		};
	},
	methods: {
		addTodoItem() {
			if (this.inputValue) {
				this.$emit('add', this.inputValue);
				this.inputValue = '';
			}
		}
	}
};
</script>

<style lang="stylus" scoped>
.header-box {
  line-height: 60px;
  background: #666;
}

.header-content {
  width: 600px;
  margin: 0 auto;
  color: #fff;
  font-size: 24px;
}

.header-input {
  float: right;
  width: 360px;
  line-height: 24px;
  margin-top: 16px;
  outline: none;
  color: #666;
  text-indent: 10px;
}
</style>
