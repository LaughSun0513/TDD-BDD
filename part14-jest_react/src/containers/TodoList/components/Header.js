import React, { Component } from "react";

export default class Header extends Component {
    state = {
        value: "",
    };
    handleInputChange(e) {
        this.setState({
            value: e.target.value,
        });
    }
    handleInputKeyUp(e) {
        const { value } = this.state;
        if (e.keyCode === 13 && value) {
            this.props.addUndoItem(value);
            this.setState({
                value : ''
            })
        }
    }
    render() {
        const { value } = this.state;
        return (
            <div className="header-box" data-test="Header">
                <div className="header-content">
                    TodoList
                    <input
                        className="header-input"
                        data-test="input"
                        placeholder = "请输入待办事项"
                        value={value}
                        onChange={this.handleInputChange.bind(this)}
                        onKeyUp={this.handleInputKeyUp.bind(this)}/>
                </div>  
            </div>
        );
    }
}
