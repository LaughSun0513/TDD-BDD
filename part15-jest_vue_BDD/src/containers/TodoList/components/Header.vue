<template>
	<div class="header-box">
		<div class="header-content">
			TodoList
			<input
				class="header-input"
				data-test="header-input"
				:value="inputValue"
				@input="e => changeInputValue(e.target.value)"
				@keyup.enter="addTodoItem"
				placeholder="请输入待办事项"
			/>
		</div>
	</div>
</template>

<script>
import { mapState, mapMutations } from 'vuex';
export default {
	name: 'Header',
	computed: {
		...mapState({
			inputValue: state => state.inputValue
		})
	},
	methods: {
		addTodoItem() {
			if (this.inputValue) {
				this.$emit('add', this.inputValue);
				this.changeInputValue('');
			}
		},
		...mapMutations(['changeInputValue'])
	}
};
</script>

<style lang="stylus" scoped>
.header-box {
  line-height: 60px;
  background: #666;
}

.header-content {
  width: 600px;
  margin: 0 auto;
  color: #fff;
  font-size: 24px;
}

.header-input {
  float: right;
  width: 360px;
  line-height: 24px;
  margin-top: 16px;
  outline: none;
  color: #666;
  text-indent: 10px;
}
</style>
