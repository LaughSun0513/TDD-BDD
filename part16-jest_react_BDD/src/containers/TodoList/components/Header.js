import React, { Component } from "react";
import { connect } from "react-redux";
import { actions } from '../store';

class Header extends Component {
    handleInputKeyUp(e) {
        const { value } = this.props;
        if (e.keyCode === 13 && value) {
            this.props.addUndoItem(value);
            this.props.handleInputChange('');
        }
    }
    render() {
        const {
            value,
            handleInputChange
        } = this.props;
        return (
            <div className="header-box" data-test="Header">
                <div className="header-content">
                    TodoList
                    <input
                        className="header-input"
                        data-test="header-input"
                        placeholder = "请输入待办事项"
                        value={value}
                        onChange = {
                            e => handleInputChange(e.target.value)
                        }
                        onKeyUp = {
                            this.handleInputKeyUp.bind(this)
                        }
                        />
                </div>  
            </div>
        );
    }
}

const mapStateToProps = state => { 
    return {
        value: state.todo.inputValue
    }
}
const mapDispatchToProps = dispatch => ({
    handleInputChange(value) {
        dispatch(actions.handleInputChange(value));
    }
})
export default connect(mapStateToProps, mapDispatchToProps)(Header)