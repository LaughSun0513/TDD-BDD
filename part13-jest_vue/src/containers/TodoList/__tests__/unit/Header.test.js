import {
  shallowMount
} from '@vue/test-utils'
import Header from '@/containers/TodoList/components/Header'
import {
  findWrapper
} from '@/containers/TodoList/utils'

describe('Header 组件测试', () => {
  it('初始化Header,包含input框', () => {
    const wrapper = shallowMount(Header)
    const input = findWrapper(wrapper, 'input').at(0)

    expect(input.exists()).toBe(true)
  })

  it('input框初始值为空', () => {
    const wrapper = shallowMount(Header)
    const inputValue = wrapper.vm.inputValue

    expect(inputValue).toBe('')
  })

  it('input框值发生变化,v-model数据跟着变', () => {
    const wrapper = shallowMount(Header)
    const input = findWrapper(wrapper, 'input').at(0)

    input.setValue('hello todolist')
    const inputValue = wrapper.vm.inputValue
    expect(inputValue).toEqual('hello todolist')
  })

  it('input框输入回车，无内容时没反应', () => {
    const wrapper = shallowMount(Header)
    const input = findWrapper(wrapper, 'input').at(0)

    input.setValue('')
    input.trigger('keyup.enter')
    const wrapperEvents = wrapper.emitted().add
    expect(wrapperEvents).toBeFalsy()
  })

  it('input框输入回车，有内容时,向外触发事件，清空inputValue', () => {
    const wrapper = shallowMount(Header)
    const input = findWrapper(wrapper, 'input').at(0)

    input.setValue('Yux')
    input.trigger('keyup.enter')

    const wrapperEvents = wrapper.emitted().add
    const inputValue = wrapper.vm.inputValue

    expect(wrapperEvents).toBeTruthy()
    expect(inputValue).toEqual('')
  })

  it('Header样式保存,变化后提示', () => {
    const wrapper = shallowMount(Header)

    expect(wrapper).toMatchSnapshot();
  })
})
