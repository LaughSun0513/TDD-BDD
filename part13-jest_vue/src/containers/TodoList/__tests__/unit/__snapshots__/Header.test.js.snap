// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header 组件测试 Header样式保存,变化后提示 1`] = `
<div class="header-box">
  <div class="header-content">
    TodoList
    <input data-test="input" placeholder="请输入待办事项" class="header-input"></div>
</div>
`;
